### PR TITLE
Test/add api tests

### PR DESCRIPTION
**Pull request by alexis-langlet**
Test part of api

### CODE REVIEW

**Review by WoodenMaiden — APPROVED** [review]
Besides this this PR looks good :+1:

File: api/src/api/service.rs [hunk 1]
```diff
@@ -75,7 +75,6 @@ impl LambdoApiService {
     }
 
     fn generate_steps(
-        &self,
```
Comment (lucido-simon) [inline api/src/api/service.rs:78]: I'm curious as to why you made this change; is there a specific reasoning ?

File: api/Cargo.toml [hunk 2]
```diff
@@ -23,6 +23,8 @@ rand = "0.8.4"
 tokio = { version = "1.0", features = ["rt-multi-thread", "macros", "process"] }
 tonic = { version = "0.10.2", features = ["transport"] }
 prost = "0.12.1"
+async-trait = "0.1.74"
```
[the remark this answers is not in the document]
Comment (lucido-simon) [reply api/Cargo.toml:26]: It's a dependency of lambdo, even if we could import it from tonic. We should not rely on reexport of other crates, unless we need to use a version that always matches the one inside the reexporting crate.

imho, this is fine

File: api/src/vm_manager/mod.rs
Comment (lucido-simon) [inline api/src/vm_manager/mod.rs:1]: Could you squash the changes on this file between 926a442 and 956b841 ? You're just overwriting what you did the commit before

**Review by lucido-simon — CHANGES_REQUESTED** [review]
Overall, LGTM, excepts a few nits.

I'm worried we are adding too much complexity to the code just to be able to test it, but I unfortunately don't see any other way of doing it.

File: api/src/api/service.rs (hunk 1 above)
Comment (alexis-langlet) [reply api/src/api/service.rs:78]: This function doesn't depend at all from &self, it doesn't use it and force to instantiate the struct to be able to call this function. 
IMO it's better to have static function whenever that's possible (and easily doable).

File: api/Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexis-langlet) [reply api/Cargo.toml:26]: @WoodenMaiden Is that ok with you if I leave it as it is ?

File: api/src/vm_manager/mod.rs
Comment (alexis-langlet) [reply api/src/vm_manager/mod.rs:1]: You're right, it's done !

**Review by lucido-simon — APPROVED** [review]
LGTM

File: api/Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (WoodenMaiden) [reply api/Cargo.toml:26]: @lucido-simon is right you can ignore the comment

**Pull request merged**